packages/types/src/Did.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import type { KiltAddress } from './Address'
+
+type AuthenticationKeyType = '00' | '01'
+type DidUriVersion = '' | `v${string}:`
+type LightDidEncodedData = '' | `:${string}`
+
+/**
+ * A string containing a KILT DID Uri.
+ */
+export type DidUri =
+  | `did:kilt:${DidUriVersion}${KiltAddress}`
+  | `did:kilt:light:${DidUriVersion}${AuthenticationKeyType}${KiltAddress}${LightDidEncodedData}`
+
+/**
+ * The fragment part of the DID URI including the `#` character.
+ */
+export type UriFragment = `#${string}`
+
+/**
+ * URL for DID resources like keys or services.
+ */
+export type DidUrl =
+  | `${DidUri}${UriFragment}`
+  // Very broad type definition, mostly for the compiler. Actual regex matching for query params is done where needed.
+  | `${DidUri}?{string}${UriFragment}`
+
+export type SignatureVerificationRelationship =
+  | 'authentication'
+  | 'capabilityDelegation'
+  | 'assertionMethod'
+export type EncryptionRelationship = 'keyAgreement'
+
+export type VerificationRelationship =
+  | SignatureVerificationRelationship
+  | EncryptionRelationship
+
+export type DidSignature = {
+  signerUrl: DidUrl
```

Comment:
what's the reason for this rename?

Reply:
You mean for the `DidSignature` field? If that's what you are referring to, switching to proper DID Documents implies, in my opinion, using also the right terminology for DID URI vs DID URL. A DID URI + a fragment for a key is a DID URL. So here I wanted to convey that what's attached to a signature is the URL of the signer (legacy name: key URI). Maybe I could rename the `signer` part to `verificationMethod`, if you so like.

Reply:
I see, so it's for consistency's sake. I was irritated a bit by it because it seemed a needless breaking change. Yes the name was not ideal, but is it worth introducing a change that will break interoperability with older sdk versions (which will not be able to verify this) for this rename? 
If we are going to rename it though, I much prefer simply calling it `verificationMethod`, aligning with the convention of LD proofs like the DataIntegrityProof

Reply:
`verificationMethod` is used to represent the whole object. This is technically a `verificationMethodId`. Would you use this instead?

Reply:
Or `verificationMethodUrl`, to distinguish it from cases where only the fragment is present. What do you think?

Reply:
just `verificationMethod`. That's what has been in use in linked data proofs and their successor, the data integrity proofs (see https://www.w3.org/TR/vc-data-integrity/#proofs). Keep in mind that JSON-LD does not care if you specify a node by value of by reference (-> id), so using a string containing the verificationMethod id instead of the actual object is perfectly normal for those people.

Reply:
Yes, I see your point, but it is still our SDK we are talking about. See [my comment on your PR](https://github.com/KILTprotocol/sdk-js/pull/801#discussion_r1370098234) for my answer to this point.

Reply:
As a web developer I’m utterly confused by this interpretation of the [difference between URL and URI](https://www.rfc-editor.org/rfc/rfc3986#section-1.1.3). I expect that others will be confused as well.

Reply:
I see. But we have to adhere to [this](https://www.w3.org/TR/did-core/#did-syntax) interpretation of URI and [this](https://www.w3.org/TR/did-core/#did-url-syntax) interpretation of URL, according to the DID Core spec.

Reply:
Right, the DID spec authors have done it again :see_no_evil:

Reply:
but to get back to the issue that I originally wanted to flag here: I'm against renaming the fields on the DID signature, as this is more of a legacy structure and appears on the old presentations as well as on sporran interfaces as far as I know.
Most things will be signed using data integrity proofs anyway in the future, so why needlessly break this?

Reply:
Ok, I'll take the breaking change argument as valid. I will also change `DidUri` to `Did` and keep `DidUrl`. Will try to change occurrences everywhere in JSDoc as well.

Reply:
Thanks for the good feedback. I have [reverted the commit for the changes in the `DidSignature` type](https://github.com/KILTprotocol/sdk-js/pull/799/commits/b2f12c5a010b77c5e70bbfe6f5069683b691d7e5) and [tried to change `DidUri` to `Did` everywhere](https://github.com/KILTprotocol/sdk-js/pull/799/commits/2fda0d66c1fa5dc0b0655a5ca98035e6921417fc). Let me know what you think.

Reply:
I personally like it!